### ISSUE TITLE
Unable to run prettier 3.0.0 with error node_modules/.bin/prettier: Permission denied

### ISSUE THREAD

**Issue by dylanhillier** [prettier/prettier#15164]
**Environments:**

- Prettier Version: 3.0.0
- Running Prettier via: pnpm, npx, etc.. doesn't appear to matter.
- Runtime: pnpm 7.14.1, node 16.16.0
- Operating System: macOS
- Prettier plugins (if any): 

**Steps to reproduce:**

Install prettier as per the docs (in my case pnpm)): 
https://prettier.io/docs/en/install.html

**Expected behavior:**

`npx prettier` should run.

**Actual behavior:**

Executing:
`npx prettier`

Yields:
sh: /Users/mrmagoo/Development/mobile/react-native-app/node_modules/.bin/prettier: Permission denied

Looking at the premissions of ./bin.prettier:

```
lrwxr-xr-x  1 mrmagoo  staff  28 Jul 27 18:12 node_modules/.bin/prettier -> ../prettier/bin/prettier.cjs
...
-rw-r--r--  2 mrmagoo  staff   2.1K Jul 25 12:55 node_modules/prettier/bin/prettier.cjs
```

If i grant execute permissions to `node_modules/prettier/bin/prettier.cjs` it all works as expected.

**Comment by fisker** [comment]
We have test for it. https://github.com/prettier/prettier/blob/6ed37ad85731e4e62d81481d40dfd98c6a675a10/tests/integration/__tests__/cli.js#L6

**Comment by dylanhillier** [comment]
Thanks for the quick response. Something funny is going on, I reinstalled it a few times this morning after removing the `node_modules` directory and it now appears to be working. However, my build pipeline is still failing with the same issue i had locally yesterday.

One additional thing i noticed was that even after the `pnpm remove prettier`, the bin file symlink is not cleaned up, it is still in `node_modules/.bin`

Will post further details if i figure out what is causing the problem.

**Comment by raduflp** [comment]
I can confirm that I've had the same issues yesterday with `prettier.cjs` not having execute permissions. 
Removing `node_modules` and reinstalling the dependencies fixed it

Prettier Version: 3.0.1
Runtime: yarn 4.0.0-rc.48, node v20.5.0
Operating System: macOS

**Comment by strongpauly** [comment]
I've seen this in multiple repos, seemingly after upgrading to prettier 3.0.x.  A workaround is to run `chmod +x ./node_modules/.bin/prettier` as OP suggests.

**Comment by Xor767** [comment]
Hi, 
same issue when you upgrade to the 3.0.1 and to 3.0.2 and the 3.0.3 version.
Any fix ongoing?

**Comment by liuyinz** [comment]
Same issue here

**Comment by fisker** [comment]
I can't reproduce on my machine, what NPM client are you using? Can someone setup a reproduction repo?

**Comment by ipanasenko** [comment]
I think this happened to me couple times after prettier got updated in our repo.
The fix was to remove `node_modules/.bin/prettier` and run `yarn` again

**Comment by ipanasenko** [comment]
I think I've managed to create reproduction repo: https://github.com/ipanasenko/prettier-3-bug-demo

To reproduce the bug:
- checkout commit `prettier works`
- run `yarn`
- run `npx prettier --check index.js` to verify that it works
- checkout commit `prettier does not work`
- run `yarn`
- run `npx prettier --check index.js` to verify that it does not work

**Comment by ipanasenko** [comment]
We didn't have this problem before recently, and I don't see it with other executables.
It's easy to check with other libraries in the same way that I did.
I've just checked eslint using same method - works fine after update

**Comment by fisker** [comment]
Yes, same problem with the repro steps in linked issue.

```js
ls -l node_modules/prettier/bin/prettier.cjs
# -rwxr-xr-x 1 ... .... 2141 Sep  4 12:59 node_modules/prettier/bin/prettier.cjs
rm -rf ./node_modules/prettier/
npm install
ls -l node_modules/prettier/bin/prettier.cjs
# -rw-r--r-- 1 ... ... 2141 Sep  4 13:02 node_modules/prettier/bin/prettier.cjs
```

**Comment by fisker** [comment]
> ```js
> ls -l node_modules/prettier/bin/prettier.cjs
> ```

You are right, it doesn't happen to eslint, I don't know what's the root cause.

**Comment by mikeharder** [comment]
I believe we are seeing a similar error in our builds.  Unfortunately, it only happens occasionally and we cannot repro on-demand.

```
 WARN  Failed to create bin at D:\a\_work\1\s\common\temp\node_modules\.bin\prettier.
 ENOENT: no such file or directory, chmod 'D:\a\_work\1\s\common\temp\node_modules\.bin\prettier'
 EPERM  EPERM: operation not permitted, open 'D:\a\_work\1\s\common\temp\node_modules\.bin\prettier.ps1'
```

We are using `prettier@3.0.3` with `@microsoft/rush@5.100.1` and `pnpm@8.6.12`.

**Comment by liuyinz** [comment]
uninstall and reinstall, solve it.

**Comment by jdelStrother** [comment]
I've been seeing this recently with prettier installed via yarn. I wonder if https://github.com/yarnpkg/berry/issues/5344 is at fault.

**Comment by kvz** [comment]
It's happening for all our berry-converted repos. Added `chmod 755 ./node_modules/.bin/prettier` to all  of their ci instructions right after `yarn --immutable` as a workaround

**Comment by KFCVme50-CrazyThursday** [comment]
This also happened when I used pnpm， I had to use sudo to install it

**Comment by matzar** [comment]
You could install prettier globally:

```shell
npm i -g prettier
```

and then use npx to run it
```shell
npx prettier --write .
```

this worked for me.

**Comment by benjie** [comment]
For me upgrading yarn to 4.9.2 and then reinstalling node_modules (`yarn set version stable && rm -R node_modules && yarn`) resolved it, probably due to this recently-merged fix:

- https://github.com/yarnpkg/berry/pull/6807

Alternatively you can fix the permissions yourself:

```sh
chmod +x node_modules/prettier/bin/prettier.cjs
```

**Comment by fisker** [comment]
Close due to unactionable.
